core/common/scale.cpp

```diff
@@ -0,0 +1,506 @@
+          if (__BYTE_ORDER == __LITTLE_ENDIAN) {
+            return value();
+          } else {
+            return reversed().value();
+          }
+        }
+
+        T value_;
+        uint8_t array_[sizeof(T)];
+      };
+
+      /**
+       * Helper class used for reinterpreting signed integers as unsigned and
+       * vice-versa
+       * @tparam T Signed or unsigned interger type
+       */
+      template <class T>
+      union SignReinterpreter {
+        using unsigned_type = typename std::make_unsigned<T>::type;
+        using signed_type = typename std::make_signed<T>::type;
+
+        SignReinterpreter<T>(unsigned_type uv) : unsigned_value_{uv} {}
+
+        SignReinterpreter<T>(signed_type sv) : signed_value_{sv} {}
+
+        unsigned_type asUnsigned() const {
+          return unsigned_value_;
+        }
+
+        signed_type asSigned() const {
+          return signed_value_;
+        }
+
+       private:
+        unsigned_type unsigned_value_;
+        signed_type signed_value_;
+      };
+
+      //      template <class T>
+      //      T encodeLittleEndian(T value) {
+      //        const int size = sizeof(T);
+      //        T result{0};
+      //        T multiplier = 1;
+      //        for (int i = 0; i < size; ++i) {
+      //          result += (value & 0xFF) * multiplier;
+      //          value >>= 8;
+      //          multiplier <<= 8;
+      //        }
+      //      }
+
+      /// Transforms host-endian- encoded integer to little-endian
+      /// \tparam T Integer type
+      /// \param value Source Integer
+      /// \return  Little-endian encoded value
+      template <class T>
+      T encodeLittleEndian(T value) {
+        return AsArray<T>(value).toLittleEndian();
+      }
+
+      /// Transforms little-endian- encoded value to host-endian
+      /// \tparam T Integer type
+      /// \param value Source integer
+      /// \return Host-endian encoded value
+      template <class T>
+      T decodeLittleEndian(T value) {
+        return AsArray<T>(value).fromLittleEndian();
+      }
+    }  // namespace impl
+
+    namespace util {
+      template <class T>
+      auto makeSignReinterpreter(T value) {
+        return impl::SignReinterpreter<T>{value};
+      }
+    }  // namespace util
+
+    namespace fixedwidth {
+      template <class T>
+      T encodePlainInteger(T value) {
+        return impl::encodeLittleEndian(value);
+      }
+
+      template <class T>
+      T decodePlainInteger(T value) {
+        return impl::decodeLittleEndian(value);
+      }
+
+      uint8_t encodeTristate(Tristate value) {
+        switch (value) {
+          case Tristate::kNone:
+            return static_cast<uint8_t>(0x0);
+          case Tristate::kFalse:
+            return static_cast<uint8_t>(0x1);
+          case Tristate::kTrue:
+            return static_cast<uint8_t>(0x2);
+          default:
+            break;
+        }
+
+        assert(false && "Value is not Tristate");
```

Comment:
asserts will be removed in debug mode, that should explicitly call std::abort or similar

Reply:
eplaced by
// code is unreachable
return 0;
this code is unreachable anyway.

Reply:
Can I consider this conversation resolved?